arcade/types.py

```diff
@@ -14,15 +14,255 @@
     Union,
     TYPE_CHECKING,
 )
+from arcade.utils import (
+    IntOutsideRangeError,
+    ByteRangeError,
+    NormalizedRangeError
+)
 from pytiled_parser import Properties
 
 if TYPE_CHECKING:
     from arcade.texture import Texture
 
 
+MAX_UINT24 = 0xFFFFFF
+MAX_UINT32 = 0xFFFFFFFF
+
+
 RGB = Tuple[int, int, int]
 RGBA = Tuple[int, int, int, int]
-Color = Union[RGB, RGBA]
+
+
+# Color = Union[RGB, RGBA]
+class Color(tuple):
```

Comment:
Is there a problem it being a NamedTuple? Might make things more informative when using it.

Reply:
I may be wrong, but to my understanding:
1. Direct subclassing of NamedTuple generates a `__new__` method which precludes boundary checking efficiently
2. We can subclass with a specific type (`Color(tuple[int, int, int, int])`) which may be friendlier to type checkers

If one or both of these are wrong, we might be able to drop the use of `# type: ignore` in `__new__` as a workaround for https://github.com/python/mypy/issues/8541.

Reply:
`NamedTuple` is also very slow